### DISCUSSION TITLE
How to extract extension value on common/native target?

### DISCUSSION THREAD

**Discussion by ShayOinif** [square/wire#2687] [Q&A]
Hi.
I have this Proto:

```protobuf
syntax = "proto3";

import "options.proto";

message MyRequest {
  string some_payload = 1 [ (my_custom_option) = "option" ];
}
```

And another Proto file defining the extension:

```protobuf
syntax = "proto3";

import "google/protobuf/descriptor.proto";

extend google.protobuf.FieldOptions {
  optional string my_custom_option = 9000;
}
```

The generated code contains an annotation class for the extension and puts the annotation on 
the field like so:

```kotlin
@Retention(AnnotationRetention.RUNTIME)
@Target(
  AnnotationTarget.PROPERTY,
  AnnotationTarget.FIELD,
)
public annotation class MyCustomOptionOption(
  public val `value`: String,
)
```

```kotlin
public class MyRequest(
  @MyCustomOptionOption("option")
  @field:WireField(
    tag = 1,
    adapter = "com.squareup.wire.ProtoAdapter#STRING",
    label = WireField.Label.OMIT_IDENTITY,
    jsonName = "somePayload",
    schemaIndex = 0,
  )
  @JvmField
  public val some_payload: String = "",
```
 
Usually on JVM I would use reflection to get the value:

```
MyRequest::some_payload.annotations.first() as MyCustomOptionOption).value
```

But this is possible only on JVM target.
How can I access the value in common/native targets?

Regards,
Shay Oinif

**Comment by ShayOinif** [comment]
Any updates on the matter?

**Reply by oldergod** [reply]
I would go through [kotlin-reflect](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect/) to see what's available. I have not done it before and don't know how that can be achieved.

**Reply by ShayOinif** [reply]
Well, reflection is quite limited in common.
On jvm I can use it to extract annotations..
